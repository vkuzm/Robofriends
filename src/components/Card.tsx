import * as React from 'react';
import { IRobot } from '../containers/App';

/*
const Card = (robot: IRobot) => {
  return (
    <div className='tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5'>
      <img alt='robots' src={`https://robohash.org/${robot.id}?200x200`} />
      <div>
        <h2>{robot.name}</h2>
        <p>{robot.email}</p>
      </div>
    </div>
  );
};
*/

const Card: React.SFC<IRobot> = ({ name, email, id }) => {
  return (
    <div className='tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5'>
      <img alt='robots' src={`https://robohash.org/${id}?200x200`} />
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
};

export default Card;