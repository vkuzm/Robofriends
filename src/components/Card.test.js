import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

it('renders without crashing', () => {
	const mockRobot = {
		id: 100500,
		name: 'Robo',
		email: 'robo@mail.com'
	}

	const shallowComponent = toJson(shallow(
		<Card 
			key={mockRobot.id} 
		  id={mockRobot.id} 
		  name={mockRobot.name} 
		  email={mockRobot.email}
		/>));

  expect(shallowComponent).toMatchSnapshot();
});