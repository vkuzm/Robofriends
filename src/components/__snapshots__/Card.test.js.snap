// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
>
  <img
    alt="Robo"
    src="https://robohash.org/100500?200x200"
  />
  <div>
    <h2>
      Robo
    </h2>
    <p>
      robo@mail.com
    </p>
  </div>
</div>
`;
